src/src/PluginStructs/P036_data_struct.cpp

```diff
@@ -425,41 +433,132 @@ void P036_data_struct::display_indicator() {
   }
 }
 
+int16_t P036_data_struct::GetHeaderHeight()
+{
+  if (bHideHeader) {
+    // no header
+    return 0;
+  }
+  else
+  {
+    return P36_HeaderHeight;
+  }
+}
+int16_t P036_data_struct::GetIndicatorTop()
+{
+  if (bHideFooter) {
+    // no footer (indicator) -> returm max. display height
+    return getDisplaySizeSettings(disp_resolution).Height;
+  }
+  else
+  {
+    return P036_IndicatorTop;
+  }
+}
+
+tFontSettings P036_data_struct::CalculateFontSettings(uint8_t lDefaultLines)
+{
+  tFontSettings result;
+  int iHeight;
+  int8_t iFontIndex = -1;
+  uint8_t iMaxHeightForFont;
+  uint8_t iLinesPerFrame;
+
+  if (lDefaultLines == 0) 
+  {
+    // number of lines can be reduced if no font fits the setting
+    iLinesPerFrame = ScrollingPages.linesPerFrame;
+    iHeight = GetIndicatorTop() - GetHeaderHeight();
+  }
+  else
+  {
+    // number of lines is fixed (e.g. for splash screen)
+    iLinesPerFrame = lDefaultLines;
+    iHeight = getDisplaySizeSettings(disp_resolution).Height;
+  }
+  
+  result.Space = 0;
+
+  while (iFontIndex < 0) {
+    iMaxHeightForFont = (iHeight - (iLinesPerFrame - 1)) / iLinesPerFrame;  // at least 1 pixel space between lines
+
+    for (uint8_t i = P36_MaxFontCount; i > 0; i--) {
+      // check available fonts for the line setting  
+      if (FontSizes[i-1].Height > iMaxHeightForFont) {
+        // height of font is to big
+        break;
+      }
+      iFontIndex = i-1; // save the current index
+      if (FontSizes[iFontIndex].Height == iMaxHeightForFont) {
+        // height of font just fits the line setting
+        break;
+      }
+    }
+    if (iFontIndex < 0) {
+      // no font fits -> rduce number of lines per page
+      iLinesPerFrame--;
+      if (iLinesPerFrame == 0)
+        break;
+    }
+  }
+  if (iFontIndex >= 0) {
+    // font found -> calculate top position and space between lines
+    iMaxHeightForFont = FontSizes[iFontIndex].Height * iLinesPerFrame;
+    if (iLinesPerFrame > 1)
+      result.Space = (iHeight-iMaxHeightForFont) / iLinesPerFrame;
```

Comment:
please add {} here to make clear what line should be depending on this if statement.
Especially since the next line does a `iLinesPerFrame-1`

Reply:
Done, see my next commit.